samples/Gaussian_initial_state/gauss_wavefcn.qs

```diff
@@ -0,0 +1,227 @@
+        for (n in -N..N) {
+            set sum = PlusD(sum, Norm_term(sigma, mu, n));
+        }
+        return sum;
+    }
+
+    /// # Summary
+    /// Computes the rotation angle.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation.
+    /// ## mu
+    /// Mean.
+    /// ## N
+    /// The limit of the sum in the normalization factor.
+    operation Angle (sigma: Double, mu: Double, N : Int) : Double {
+        return ArcCos(Sqrt(Norm(sigma/2., mu/2., N)/Norm(sigma, mu, N)));
+    }
+
+    /// # Summary
+    /// Return a list of n-bit strings.
+    /// # Input
+    /// ## n
+    /// The number of bits.
+    operation QubitStrings (n: Int) : Bool[][] {
+        mutable array = ConstantArray(2^n, IntAsBoolArray(0,n));
+        for (i in 0..2^n - 1) {
+            let bitstring = IntAsBoolArray(i,n);
+            set array w/= i <- bitstring;
+        }
+        return array;
+    }
+
+    
+    /// # Summary
+    /// Given an n-bit string, return the corresponding mean for the rotation angle
+    /// at recursion level n.
+    /// # Input
+    /// ## qub
+    /// The n-bit string.
+    /// ## mu
+    /// Mean.
+    operation MeanQubitCombo (qub: Bool[], mu: Double) : Double {
+        mutable mu_out = mu;
+        for (bit in qub) {
+            mutable i = 0.0;
+            if (bit == true) {
+                set i = 1.;
+            }
+            set mu_out = mu_out/2. - i/2.;
+        }
+        return mu_out;
+    }
+
+    /// # Summary
+    /// At recursion level n, return a list of all the means used for the various rotation angles.
+    /// # Input
+    /// ## mu
+    /// Mean.
+    /// ## n
+    /// Recursion level.
+    operation LevelMeans (mu: Double, n: Int) : Double[] {
+        mutable list_mu_out = ConstantArray(2^n, 0.);
+        let qb_strings = Qubit_strings(n);
+        for (i in 0..2^n-1) {
+            let mu_out = MeanQubitCombo(ElementAt(i, qb_strings), mu);
+            set list_mu_out w/= i <- mu_out;
+        }
+        return list_mu_out;
+    }
+
+    
+    /// # Summary
+    /// At recursion level n, return a list of all the rotation angles.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation
+    /// ## mu
+    /// Mean.
+    /// ## n
+    /// Recursion level.
+    operation LevelAngles(sigma: Double, mu: Double, n: Int) : Double[] {
+        let sigma_out = sigma/(2.^IntAsDouble(n));
+        let list_mu = Level_means(mu, n);
+        mutable angles_out = ConstantArray(2^n, 0.);
+        for (i in IndexRange(list_mu)) {
+            mutable mu_ = ElementAt(i, list_mu);
+            set angles_out w/= i <- Angle(sigma_out, mu_, 10^3);
+        }
+        return angles_out;
+    }
+
+    /// # Summary
+    /// Create and return the Gaussian wavefunction.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation
+    /// ## mu_
+    /// Mean.
+    /// ## num_qubits
+    /// The number of qubits.
+    operation GaussWavefcn (sigma: Double, mu_: Double, numQubits: Int) : Unit {
```

Comment:
Changed! Shall I make PrepareGaussWavefcnRecursive also adjointable? Thanks!

Reply:
And I also have a Python file host.py for running the program. I'm not sure if I can allocate register in the Python file. Shall I just delete the Python file? Thanks!

Reply:
Or we can make a separate convenience operation, then we can keep the Python file? Thanks!

Reply:
> And I also have a Python file host.py for running the program. I'm not sure if I can allocate register in the Python file. Shall I just delete the Python file? Thanks!

You're right, quantum registers can't be allocated from host programs. If you think of a Q# program as being something you send to quantum hardware to run a job, then the host program is what's responsible for submitting that job — that is, the host program runs on a conventional classical machine and not on the actual or simulated quantum hardware.

Anyway, there's definitely nothing wrong with having a Python host program, we love to be able to demonstrate how Q# and Python work together! In this case, however, `host.py` doesn't really do much beyond what the standalone Q# project already does, such that as you say, there may not be as much point keeping it as a separate program. Perhaps folding it into `read_state.py`, so that it both dispatches the Q# program to the simulator and reads out the dump?

(Since `DumpMachine` outputs are intended for human readability and not machine parsing, ideally the host would intercept the diagnostic generated by `DumpMachine` instead of parsing, but we've not really documented how to do that yet unfortunately... that would be a great feature request, though!)

Reply:
Thanks, @cgranade, this is a great suggestion. I've discussed with Andy and he's going to consolidate the Python scripts and refactor the entrypoint to take input arguments.